invenio_app_rdm/theme/assets/js/react_invenio_deposit/components/ResourceTypeField.js

```diff
@@ -8,45 +8,49 @@
 import React, { Component } from "react";
 import PropTypes from "prop-types";
 import { getIn, Field } from "formik";
+import { Form, Icon } from "semantic-ui-react";
 
 import { SelectField } from "../../react_invenio_forms";
 
+
 export class ResourceTypeField extends Component {
   renderResourceTypeField = (formikBag) => {
-    const resource_type_fieldpath = `${this.props.fieldPath}.type`;
-    const resource_subtype_fieldpath = `${this.props.fieldPath}.subtype`;
+    const typeFieldPath = `${this.props.fieldPath}.type`;
+    const subtypeFieldPath = `${this.props.fieldPath}.subtype`;
 
     const resource_type = getIn(
       formikBag.form.values,
-      resource_type_fieldpath,
+      typeFieldPath,
       ""
     );
 
     const handleChange = (event, selectedOption) => {
       formikBag.form.setFieldValue(
-        resource_type_fieldpath,
+        typeFieldPath,
         selectedOption.value
       );
-      formikBag.form.setFieldValue(resource_subtype_fieldpath, "");
+      formikBag.form.setFieldValue(subtypeFieldPath, "");
     };
 
-    const subtype_options = this.props.options.subtype.filter(
+    const subtypeOptions = this.props.options.subtype.filter(
       (e) => e["parent-value"] === resource_type
     );
     return (
       <div>
+        <label><Icon name={this.props.labelIcon} />{this.props.label}</label>
```

Comment:
@fenekku Here, it seems that the icon is a required prop but not defined as one in the propTypes section. Should we require the use of an icon, do you think? Is this coming directly from the vocabulary endpoint?

Reply:
The icon shouldn't be required. I wasn't getting errors when `labelIcon` wasn't passed, but I now I see that the `Icon` component still renders which takes whitespace/looks strange. I will make it not appear if `labelIcon` is not passed to look better :).